yt_dlp/extractor/flextv.py

```diff
@@ -0,0 +1,61 @@
+from .common import InfoExtractor
+from ..networking.exceptions import HTTPError
+from ..utils import (
+    ExtractorError,
+    parse_iso8601,
+    str_or_none,
+    traverse_obj,
+    UserNotLive,
+    url_or_none,
+)
+
+
+class FlexTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?flextv\.co\.kr/channels/(?P<id>\d+)/live'
+    _TESTS = [{
+        'url': 'https://www.flextv.co.kr/channels/231638/live',
+        'info_dict': {
+            'id': '231638',
+            'ext': 'mp4',
+            'title': r're:^214하나만\.\.\. ',
+            'thumbnail': r're:^https?://.+\.jpg',
+            'upload_date': r're:\d{8}',
+            'timestamp': int,
+            'live_status': 'is_live',
+            'channel': 'Hi별',
+            'channel_id': '244396',
```

Comment:
why does this channel_id differ from the url channel_id?

Reply:
That's how their API works.
Here is the full response:

<details>

```json
{
	"id": 231638,
	"title": "Hi 오빠들 ❤️",
	"description": "월목 휴방 ❤️ \r\nAM 9시 시작 ❤️",
	"thumbUrl": "https://ivs.flextv.co.kr/ivs/v1/585677956555/ntFH812F7k4G/2024/2/17/23/25/0TbKtHNWtsUf/media/latest_thumbnail/thumb.jpg",
	"fanIcons": [
		{
			"id": 3012,
			"channelId": 231638,
			"type": "RANKING",
			"value": "1",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_0.png"
		},
		{
			"id": 3013,
			"channelId": 231638,
			"type": "RANKING",
			"value": "2",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_1.png"
		},
		{
			"id": 3014,
			"channelId": 231638,
			"type": "RANKING",
			"value": "3",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_2.png"
		},
		{
			"id": 3015,
			"channelId": 231638,
			"type": "RANKING",
			"value": "fever",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_3.png"
		},
		{
			"id": 3016,
			"channelId": 231638,
			"type": "LEVEL",
			"value": "5",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_4.png"
		},
		{
			"id": 3017,
			"channelId": 231638,
			"type": "LEVEL",
			"value": "4",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_5.png"
		},
		{
			"id": 3018,
			"channelId": 231638,
			"type": "LEVEL",
			"value": "3",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_6.png"
		},
		{
			"id": 3019,
			"channelId": 231638,
			"type": "LEVEL",
			"value": "2",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_7.png"
		},
		{
			"id": 3020,
			"channelId": 231638,
			"type": "LEVEL",
			"value": "1",
			"url": "https://flex-images.kr.object.ncloudstorage.com/fanIcon/06/fanIcon_1686114918_8.png"
		}
	],
	"subBadge": [
		{
			"id": 62,
			"status": 2,
			"channelId": 0,
			"imgUrl": "https://static.flextv.co.kr/assets/Default/badge_default_icon.png",
			"month": 1,
			"createdAt": "2024-01-24T06:43:42.000Z",
			"updatedAt": null
		}
	],
	"altThumbUrl": null,
	"sources": [
		{
			"resolution": 0,
			"url": "https://66368cdf63a9.ap-northeast-2.playback.live-video.net/api/video/v1/ap-northeast-2.585677956555.channel.ntFH812F7k4G.m3u8?token=eyJhbGciOiJFUzM4NCIsInR5cCI6IkpXVCJ9.eyJhd3M6Y2hhbm5lbC1hcm4iOiJhcm46YXdzOml2czphcC1ub3J0aGVhc3QtMjo1ODU2Nzc5NTY1NTU6Y2hhbm5lbC9udEZIODEyRjdrNEciLCJleHAiOjE3MDgyNTU2MTY4NzAsImlhdCI6MTcwODIxMjQxNn0.Y7DBm-Vharc1KHhXwAvUBVS36kBwR5TcxD8inOXsuR6CyBhaqVJ0v-0w3OrQ7klfQZZ7Y0rV-lCz9r3gAhR08NdCaEnU7MBou5smJRH9KpcxYOEIzwkfr6Us_ycufATz",
			"format": "ivs"
		}
	],
	"sourceType": "aws",
	"owner": {
		"loginId": "godys9797",
		"nickname": "Hi별",
		"name": "Hi별",
		"thumbUrl": "https://jcmkeuseuzeq6748407.cdn.ntruss.com/members/244396/1bdh71lfitng5m.jpg?type=w&w=180&quality=90",
		"oauth": null,
		"fan": {
			"rating": 0
		},
		"luv": 0,
		"role": "V",
		"sms": 1,
		"id": 244396,
		"condition": 60,
		"phone": "",
		"status": 1,
		"isAdult": true,
		"PI": {
			"isAdult": true
		},
		"ranker": false,
		"memberDiv": 3,
		"appPush": 1,
		"msgCount": 0,
		"signature": false,
		"billboardIs": {
			"billboardIsView": 1,
			"billboardIsVibes": 1,
			"billboardIsSponser": 1
		},
		"isBigFan": 1,
		"isAdultCheck": 1,
		"setting": {},
		"channelStop": null,
		"anniversary": {
			"memberBirthday": null,
			"isBirthday": false,
			"birthDday": null,
			"streamStartDateAt": null,
			"streamStartDday": null,
			"isAnniversary": false,
			"anniversaryDays": []
		},
		"oauthRegPath": null,
		"passwordCheck": null
	},
	"stream": {
		"id": 270223,
		"channelId": 231638,
		"themeId": 5,
		"title": "214하나만...",
		"resolution": 1080,
		"contentHeight": 1080,
		"contentWidth": 1920,
		"streamAgent": "PC",
		"suspended": 0,
		"blind": 0,
		"isRecording": 1,
		"isForAdult": 0,
		"password": "",
		"minRatingLevel": 0,
		"maxViewerCount": 50000,
		"useAltThumbUrl": 0,
		"userAgentType": "M_APP_V1_IVS",
		"disconnectedAt": null,
		"endedAt": null,
		"createdAt": "2024-02-17T23:25:43.000Z",
		"orderNo": 0,
		"resetdateAt": "2024-02-17T23:25:43.000Z",
		"visitAllow": 0,
		"useGame": 0,
		"stopMember": null,
		"isFrozen": false
	},
	"isMobile": false,
	"streamStats": {
		"topFanId": "27711"
	},
	"role": {
		"app": "V",
		"channel": "V"
	},
	"openUrl": {
		"chat": "/redirects/signin?token=undefined&redirectTo=/popup/chat/231638",
		"chatMobile": "/redirects/signin?token=undefined&redirectTo=/popup/m/chat/231638"
	},
	"status": {
		"stat": {
			"numberOfViews": 0,
			"numberOfSponsor": 0,
			"numberOfRecommendations": 0,
			"numberOfSubs": 313,
			"totalGrade": 0
		},
		"barrier": {
			"blind": 0,
			"suspended": false,
			"isLocked": false,
			"minRatingLevel": 0,
			"isForAdult": 0
		},
		"title": "214하나만...",
		"themeId": 5,
		"startedAt": "2024-02-17T23:25:43.000Z",
		"resolution": 1080,
		"password": "",
		"visitAllow": 0,
		"useHigh": 1,
		"tag": []
	},
	"broadcastNotice": "Hi별이 명령어\n\n!프로필 / !미션 / !댄스 / !1분댄스 / !노래 / !치사 / !얼낙\n\n열심히 하겠습니다!♡\n",
	"broadcastRule": null,
	"channelExternalLink": null
}
```
</details>